lib/src/cli/install/install_command.dart

```diff
@@ -237,195 +125,82 @@ class InstallCommand extends Command<void> {
-
-    final dowloadedBinaryFile = File(path.join(destinationDir.absolute.path, binaryName));
-    print("Copying realm binary ${dowloadedBinaryFile.absolute.path} to ${binaryFile.absolute.path}");
-    await dowloadedBinaryFile.copy(binaryFile.absolute.path);
-  }
-
-  Future<void> downloadAndExtractLinuxBinaries(String realmPackagePath, Pubspec realmPubspec) async {
-    final destinationDir = Directory(path.join(Directory.systemTemp.absolute.path, "realm-binary", "linux"));
-    final archiveName = "linux.tar.gz";
-    final destinationFile = File(path.join(destinationDir.absolute.path, archiveName));
-    if (!await skipDownload(destinationDir.absolute.path, realmPubspec.version.toString())) {
-      await download(destinationFile, realmPubspec, archiveName);
-    }
-
-    print("Extracting Realm binaries to ${destinationDir.absolute.path}");
-    final archive = Archive();
-    await archive.extract(destinationFile, destinationDir);
-    await saveVersionFile(destinationDir, realmPubspec);
-
-    final binaryName = "librealm_dart.so";
-    final binaryFile = File(path.join(Directory.current.absolute.path, "binary", "linux", binaryName));
-    //create parent dirs if needed
-    await binaryFile.create(recursive: true);
-
-    final dowloadedBinaryFile = File(path.join(destinationDir.absolute.path, binaryName));
-    print("Copying realm binary ${dowloadedBinaryFile.absolute.path} to ${binaryFile.absolute.path}");
-    await dowloadedBinaryFile.copy(binaryFile.absolute.path);
-  }
-
-  Future<void> download(File destinationFile, Pubspec realmPubspec, String archiveName) async {
-    if (!await destinationFile.exists()) {
-      await destinationFile.create(recursive: true);
-    }
-
-    print("Downloading Realm binaries for $packageName@${realmPubspec.version} to ${destinationFile.absolute.path}");
-    final client = HttpClient();
-    try {
-      var url = 'https://github.com/realm/realm-dart/releases/download/${realmPubspec.version}/$archiveName';
-      if (debug) {
-        url = 'http://localhost:8000/$archiveName';
-      }
-      final request = await HttpClient().getUrl(Uri.parse(url));
-      final response = await request.close();
-      if (response.statusCode >= 400) {
-        throw Exception("Error downloading Realm binaries from $url. Error code: ${response.statusCode}");
-      }
-      await response.pipe(destinationFile.openWrite());
-    }
-    //TODO: This does not handle download errors.
-     finally {
-      client.close(force: true);
-    }
-  }
-
-  Future<void> saveVersionFile(Directory destinationDir, Pubspec realmPubspec) async {
-    var versionFile = File(path.join(destinationDir.absolute.path, versionFileName));
+    final versionFile = File(path.join(destinationDir.absolute.path, _versionFileName));
     await versionFile.writeAsString("${realmPubspec.version}");
   }
 
-  Future<String> getRealmPackagePath() async {
+  Future<String> _getRealmPackagePath() async {
```

Comment:
Making all these methods and fields private only hurts readability imo. There no reason to hide these since we are not using this class anywhere else so we need to care of its public interface.

Reply:
I agree - I was wondering what the best practice was around this. The main reason why I did it was that there was some unused method that wasn't flagged by the compiler because it was public. But now that they're all used, I can make them public again.